be/src/olap/base_compaction.cpp

```diff
@@ -81,16 +84,61 @@ Status BaseCompaction::execute_compact_impl() {
     return Status::OK();
 }
 
+bool BaseCompaction::_filter_input_rowset() {
+    bool exceed = false;
+    // if enable dup key skip big file and no delete predicate
+    // we skip big files too save resources
+    if (!config::enable_dup_key_base_compaction_skip_big_file ||
+        _tablet->keys_type() != KeysType::DUP_KEYS || _tablet->delete_predicates().size() != 0) {
+        return false;
```

Comment:
Could we check version of delete predicates? e.g. If  we dont have delete predicates before version 10, then we can skip large rowset before version 10.